lib/mix/lib/mix/task.ex

```diff
@@ -360,6 +375,15 @@ defmodule Mix.Task do
     end
   end
 
+  def run_requirements(module) do
+    if requirements = requirements(module) do
+      Enum.each(requirements, fn requirement ->
+        [task | args] = String.split(requirement)
+        Mix.Task.run(task, args)
+      end)
+    end
```

Comment:
If we go with the empty list of requirements as the default, this would change and the `if` would be gone:

```suggestion
    Enum.each(requirements(module), fn requirement ->
      [task | args] = String.split(requirement)
      Mix.Task.run(task, args)
    end)
```

I personally think this is a nice improvement but it's not a blocking thing, we can ship this PR regardless :)